### PR TITLE
feat(ci): add changelog check to github actions and small improvments

### PR DESCRIPTION
**Pull request by vdovhanych**
This adds a changelog check to GitHub actions for external pull requests. I also adjusted the naming of other checks while on it. I can do it in a separate commit if that would be better. 

Resoles #1651

### CODE REVIEW

**Review by matejcik — APPROVED** [review]
utACK

File: .github/workflows/prebuild.yml [hunk 1]
```diff
@@ -3,19 +3,20 @@ name: "Prebuild checks"
...
   style_check:
+    name: Style check
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2.3.4
       - uses: cachix/install-nix-action@v13
         with:
           nix_path: nixpkgs=channel:nixos-unstable
-        name: "Run script"
+        name: "Install poetry and run style check"
```
Comment (tsusanka) [inline .github/workflows/prebuild.yml:14]: I would drop the "Install poetry" as it is a prerequisite not something we test per-se.

**Review by tsusanka — COMMENTED** [review]
One nit but LGTM. So we do this because external PRs do not run in Gitlab right? @matejcik shouldn't we also remove the check from Gitlab then? So we do not have it twice.

**Comment by matejcik** [comment]
> shouldn't we also remove the check from Gitlab then

style, defs and gen checks also run in gitlab.
I'd keep it, we also use gitlab internally, it can be useful to get the changelog check before you create a PR

**Comment by tsusanka** [comment]
Ok!

File: .github/workflows/prebuild.yml (hunk 1 above)
Comment (vdovhanych) [reply .github/workflows/prebuild.yml:14]: Ok removed the install poetry and left only the description "Run style check" or other checks.

**Review by mmilata — APPROVED** [review]
LGTM! The script uses the value of `$CI_COMMIT_BRANCH` which I'm not sure github actions set. However when it's not set the script does the right thing (assumes it's not a release branch), so I guess that's ok.

**Comment by tsusanka** [comment]
@mmilata We could use [GITHUB_REF](https://docs.github.com/en/actions/reference/environment-variables) for that. But I agree as that is solely for the release branch (and we do not file PR there) we do not have to worry about it..

**Pull request merged**